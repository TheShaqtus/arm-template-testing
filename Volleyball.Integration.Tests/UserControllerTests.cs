using System.Net.Http;
using System.Text;
using System.Threading.Tasks;
using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using ApprovalTests.Reporters.TestFrameworks;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Newtonsoft.Json;
using Volleyball.Models;

namespace Volleyball.Integration.Tests
{
    [UseReporter(typeof(DiffReporter), typeof(MsTestReporter))]
    [UseApprovalSubdirectory("Approvals")]
    [TestClass]
    public class UserControllerTests : TestBase
    {
        [TestMethod]
        public async Task Get()
        {
            // Arrange
            var client = new HttpClient();

            // Act
            var result = await client.GetAsync($"{Initialize.BaseUrl}/users")
                .ConfigureAwait(false);
            var users = await result.Content.ReadAsStringAsync().ConfigureAwait(false);

            // Assert
            Approvals.VerifyJson(users);
        }

        [TestMethod]
        public async Task Post()
        {
            // Arrange
            var newUser = new CreateUserModel {FirstName = "Test", LastName = "User"};

            var client = new HttpClient();
            
            var content = new StringContent(JsonConvert.SerializeObject(newUser), Encoding.UTF8, "application/json");

            // Act
            var result = await client.PostAsync(
                    $"{Initialize.BaseUrl}/users",
                    content)
                .ConfigureAwait(false);

            Assert.IsTrue(result.IsSuccessStatusCode);

            var getResults = await client
                .GetAsync($"{Initialize.BaseUrl}/users")
                .ConfigureAwait(false);

            var users = await getResults.Content.ReadAsStringAsync().ConfigureAwait(false);

            // Assert
            Approvals.VerifyJson(users);
        }
    }
}
