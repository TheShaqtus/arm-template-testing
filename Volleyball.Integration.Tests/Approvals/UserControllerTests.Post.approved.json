[
    {
        "id": 1,
        "firstName": "Test",
        "lastName": "User"
    },
    {
        "id": 2,
        "firstName": "Test",
        "lastName": "User"
    }
]